pkg/controller/elasticsearch/client/base.go

```diff
@@ -66,10 +69,15 @@ func (c *baseClient) doRequest(context context.Context, request *http.Request) (
 
 	response, err := c.HTTP.Do(withContext)
 	if err != nil {
-		return response, err
+		return response, newDecoratedHTTPError(request, err)
+	}
+
+	// Check HTTP code in Elasticsearch response.
+	if response.StatusCode < 200 || response.StatusCode >= 300 {
+		return response, newDecoratedHTTPError(request, newAPIError(response))
```

Comment:
Considering this is the only usage of APIError, and we wrap it in a `newDecoratedHTTPError`, would it be better to add the request `path` to the `APIError` struct?
So the `path` belongs to a structured object rather than a raw string.